pkg/virt-api/rest/subresource_test.go

```diff
@@ -519,6 +532,212 @@ var _ = Describe("VirtualMachineInstance Subresources", func() {
 			Expect(response.StatusCode()).To(Equal(http.StatusAccepted))
 			close(done)
 		})
+
+		Context("Subresource api - rename", func() {
+			var (
+				vm      *v1.VirtualMachine
+				newName string
+			)
+
+			newRenameBody := func(newName string) io.ReadCloser {
+				newNameBytes := []byte(newName)
+				return &readCloserWrapper{bytes.NewReader(newNameBytes)}
+			}
+
+			BeforeEach(func() {
+				vm = newMinimalVM("renametest")
+				vm.Namespace = "renametestns"
+
+				request.PathParameters()["name"] = vm.Name
+				request.PathParameters()["namespace"] = vm.Namespace
+
+				newName = vm.Name + "new"
+			})
+
+			Context("Without VM mocking", func() {
+				It("should fail if no name is provided in the request body", func(done Done) {
+					app.RenameVMRequestHandler(request, response)
+
+					status := ExpectStatusErrorWithCode(recorder, http.StatusBadRequest)
+					Expect(status.Error()).To(ContainSubstring("no body"))
+
+					close(done)
+				})
+
+				It("should fail if the new name is empty", func(done Done) {
+					request.Request.Body = newRenameBody("")
+
+					app.RenameVMRequestHandler(request, response)
+
+					status := ExpectStatusErrorWithCode(recorder, http.StatusBadRequest)
+					Expect(status.Error()).To(ContainSubstring("empty"))
+					close(done)
+				})
+
+				It("should fail if the new name is identical to the current name", func(done Done) {
+					request.Request.Body = newRenameBody(vm.Name)
+
+					app.RenameVMRequestHandler(request, response)
+
+					status := ExpectStatusErrorWithCode(recorder, http.StatusBadRequest)
+					Expect(status.Error()).To(ContainSubstring("identical"))
+
+					close(done)
+				})
+			})
+
+			Context("With source VM mocking", func() {
+				BeforeEach(func() {
+					vmGetStatus := http.StatusOK
+					request.Request.Body = newRenameBody(newName)
+
+					server.AppendHandlers(
+						ghttp.CombineHandlers(
+							ghttp.VerifyRequest("GET", getVMPath("v1alpha3", vm.Namespace, vm.Name)),
+							ghttp.RespondWithJSONEncodedPtr(&vmGetStatus, vm),
+						),
+					)
+				})
+
+				Context("With no running/runStrategy applied", func() {
+					It("should fail if no running state or runStrategy are applied", func(done Done) {
```

Comment:
Is this actually possible? I thought there was a validating webhook preventing a VM from being posted if neither Running nor RunStrategy were defined. If you've found a way to make this state actually happen, that's a bug.

Reply:
Yes, there is a validating webhook that checks for that, But because I was checking for running/runStrategy separately I wanted to make sure it works.
But now after you informed me about `vm.RunStrategy()`, this can be removed